### PR TITLE
eip_create/eip-show: support for healthchecks

### PR DESCRIPTION
**Pull request by mcorbin**
- in eip_create.go, add new parameters for the healthcheck definition
- in eip_show.go, show the healthcheck parameters.

```
./exo -A dev eip create --mode http --path "/" --port 8080
┼──────────┼─────────────────┼──────────────────────────────────────┼
│   ZONE   │       IP        │                  ID                  │
┼──────────┼─────────────────┼──────────────────────────────────────┼
│ ch-gva-2 │ 159.100.241.213 │ f0b14bcc-cd82-4364-b36d-b63d2e05af8d │
┼──────────┼─────────────────┼──────────────────────────────────────┼


./exo -A dev eip show f0b14bcc-cd82-4364-b36d-b63d2e05af8d
┼──────────┼─────────────────┼─────────────────┼────────────────────┼──────────┼──────┼──────┼──────┼──────────────┼────────────┼─────────┼
│   ZONE   │       IP        │ VIRTUAL MACHINE │ VIRTUAL MACHINE ID │ INTERVAL │ MODE │ PATH │ PORT │ STRIKES FAIL │ STRIKES OK │ TIMEOUT │
┼──────────┼─────────────────┼─────────────────┼────────────────────┼──────────┼──────┼──────┼──────┼──────────────┼────────────┼─────────┼
│ ch-gva-2 │ 159.100.241.213 │                 │                    │ 10       │ http │ /    │ 8080 │ 3            │ 2          │ 2       │
┼──────────┼─────────────────┼─────────────────┼────────────────────┼──────────┼──────┼──────┼──────┼──────────────┼────────────┼─────────┼

```

This PR needs https://github.com/exoscale/egoscale/pull/396 merged.

Story: [EIP health checking in exo CLI](https://app.clubhouse.io/exoscale/story/1915)

### CODE REVIEW

File: cmd/eip_create.go [hunk 1]
```diff
@@ -18,17 +18,55 @@ var eipCreateCmd = &cobra.Command{
 		if len(args) >= 1 {
 			zone = args[0]
 		}
-		return associateIPAddress(zone)
+
+		interval, err := cmd.Flags().GetInt64("interval")
```
[the remark this answers is not in the document]
Comment (mcorbin) [reply cmd/eip_create.go]: Updated.

**Comment by mcorbin** [comment]
@falzm What do you think about this format ?

```
┼──────────────────────────────────────┼─────────────────────────────────────────┼
│           159.100.241.247            │                                         │
┼──────────────────────────────────────┼─────────────────────────────────────────┼
│ ID                                   │ 5db34b52-5125-4688-a069-015f456c9220    │
│ Zone                                 │ ch-gva-2                                │
│                                      │                                         │
│ Healthcheck                          │                                         │
│ Mode                                 │ http                                    │
│ Path                                 │ /                                       │
│ Port                                 │ 8080                                    │
│ Interval                             │ 10                                      │
│ Strikes Fail                         │ 3                                       │
│ Strikes OK                           │ 2                                       │
│ Timeout                              │ 2                                       │
│                                      │                                         │
│ Associated Virtual Machines          │                                         │
│ e98b439a-bf89-4f9f-bf17-6146f65356a2 │ aaa                                     │
│ de6713ef-62b4-4bfb-b4dd-0ce28026c680 │ VM-de6713ef-62b4-4bfb-b4dd-0ce28026c680 │
┼──────────────────────────────────────┼─────────────────────────────────────────┼
```

File: cmd/eip_create.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (falzm) [reply cmd/eip_create.go]: 👍

**Comment by falzm** [comment]
> @falzm What do you think about this format ?

Not sure to like this: I find it confusing, and not consistent with the other formats we use in other sections of the CLI :/

File: cmd/eip_create.go [hunk 2]
```diff
@@ -18,17 +18,55 @@ var eipCreateCmd = &cobra.Command{
...
+		strikesOK, err := cmd.Flags().GetInt64("healthcheck-strikes-ok")
+		if err != nil {
+			return err
+		}
+		timeout, err := cmd.Flags().GetInt64("healthcheck-timeout")
+		if err != nil {
+			return err
+		}
+
+		return associateIPAddress(interval, port, strikesFail, strikesOK, timeout, mode, path, zone)
```
Comment (pierre-emmanuelJ) [inline cmd/eip_create.go]: What about to give an `egoscale.AssociateIPAddress` in param and fill zone id after.
better visibility and less parameters

**Comment by mcorbin** [comment]
@falzm New format:

```
┼──────────────────────────┼─────────────────────────────────────────┼
│     159.100.241.247      │                                         │
┼──────────────────────────┼─────────────────────────────────────────┼
│ ID                       │ 5db34b52-5125-4688-a069-015f456c9220    │
│ Zone                     │ ch-gva-2                                │
│ Healthcheck Mode         │ http                                    │
│ Healthcheck Path         │ /                                       │
│ Healthcheck Port         │ 8080                                    │
│ Healthcheck Interval     │ 10                                      │
│ Healthcheck Strikes Fail │ 3                                       │
│ Healthcheck Strikes OK   │ 2                                       │
│ Healthcheck Timeout      │ 2                                       │
│ Instances                │ aaa                                     │
│                          │ VM-de6713ef-62b4-4bfb-b4dd-0ce28026c680 │
┼──────────────────────────┼─────────────────────────────────────────┼

```

**Comment by falzm** [comment]
> @falzm New format:
> 
> ```
> ┼──────────────────────────┼─────────────────────────────────────────┼
> │     159.100.241.247      │                                         │
> ┼──────────────────────────┼─────────────────────────────────────────┼
> │ ID                       │ 5db34b52-5125-4688-a069-015f456c9220    │
> │ Zone                     │ ch-gva-2                                │
> │ Healthcheck Mode         │ http                                    │
> │ Healthcheck Path         │ /                                       │
> │ Healthcheck Port         │ 8080                                    │
> │ Healthcheck Interval     │ 10                                      │
> │ Healthcheck Strikes Fail │ 3                                       │
> │ Healthcheck Strikes OK   │ 2                                       │
> │ Healthcheck Timeout      │ 2                                       │
> │ Instances                │ aaa                                     │
> │                          │ VM-de6713ef-62b4-4bfb-b4dd-0ce28026c680 │
> ┼──────────────────────────┼─────────────────────────────────────────┼
> ```

LGTM 👍  Please make sure the `Healthcheck *` entries are not displayed unless there is an actual healthcheck defined.

File: go.mod [hunk 3]
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/cenkalti/backoff v2.0.0+incompatible
 	github.com/cpuguy83/go-md2man v1.0.8 // indirect
 	github.com/dustin/go-humanize v0.0.0-20180713052910-9f541cc9db5d
-	github.com/exoscale/egoscale v0.14.3
+	github.com/exoscale/egoscale v0.14.4-0.20190401095233-3ad41b3e71e6
```
Comment (falzm) [inline go.mod]: You can use `github.com/exoscale/egoscale v0.15.0` now

Comment (mcorbin) [reply go.mod]: done ;)

**Pull request merged**